servicetest/devicenode/test_devicenode_gateway.py

```diff
@@ -0,0 +1,188 @@
+    Container.READONLY: False
+}
+
+
+GW_CONFIG_SAME_DEV_0 = [
+    {
+        "name": "/dev/input/mouse0",
+        "mode": 644
+    }
+]
+
+GW_CONFIG_SAME_DEV_1 = [
+    {
+        "name": "/dev/input/mouse0",
+        "mode": 754
+    }
+]
+
+GW_CONFIG_NEW_DEV_0 = [
+    {
+        "name": "/dev/tty0",
+        "mode": 644
+    }
+]
+
+GW_CONFIG_NEW_DEV_1 = [
+    {
+        "name": "/dev/tty1",
+        "mode": 755
+    }
+]
+
+GW_CONFIG_SAME_DEV_SAME_MODE = [
+    {
+        "name": "/dev/tty0",
+        "mode": 644
+    }
+]
+
+""" Following capabilities are fed to DynamicConfiguration tests to examine the case
+    when new mode is configured for same device.
+"""
+test_cap_0 = Capability("double.activation.same.device.0.cap",
+                        [
+                            {"id": "devicenode", "config": GW_CONFIG_SAME_DEV_0}
+                        ])
+
+test_cap_1 = Capability("double.activation.same.device.1.cap",
+                        [
+                            {"id": "devicenode", "config": GW_CONFIG_SAME_DEV_1}
+                        ])
+
+""" Following capabilities are fed to DynamicConfiguration tests to examine the case
+    when a new device is added to configuration
+"""
+test_cap_2 = Capability("double.activation.new.device.0.cap",
+                        [
+                            {"id": "devicenode", "config": GW_CONFIG_NEW_DEV_0}
+                        ])
+
+test_cap_3 = Capability("double.activation.new.device.1.cap",
+                        [
+                            {"id": "devicenode", "config": GW_CONFIG_NEW_DEV_1}
+                        ])
+
+""" Following capability is fed to DynamicConfiguration tests to examine the case
+    when exactly same device with same mode is added
+"""
+test_cap_4 = Capability("double.activation.same.device.same.mode.cap",
+                        [
+                            {"id": "devicenode", "config": GW_CONFIG_SAME_DEV_SAME_MODE}
+                        ])
+
+""" Define the manifest so it can be used by the testframework  """
+manifest = StandardManifest(TESTOUTPUT_DIR,
+                            "devicenode-test-manifest.json",
+                            [
+                                test_cap_0,
+                                test_cap_1,
+                                test_cap_2,
+                                test_cap_3,
+                                test_cap_4
+                            ])
+
+
+def service_manifests():
+    """ The agent fixture calls this function when it creates the service manifests
+        that should be used with this test module. The agent fixture expects a list
+        of StandardManifest and/or DefaultManifest objects.
+    """
+    return [manifest]
+
+# Test suites
+
+
+@pytest.mark.usefixtures("testhelper", "agent")
+class TestDeviceNode(object):
+    """ This suite tests the behavior of DeviceNodeGateway within given set of capabilities.
+        It uses the DeviceNodeGateway to assert behavior.
```

Comment:
But in what way does it "use DeviceNodeGateway to assert behavior"? I know enough of the context to interpret this, but I'm not sure I will be able to do that in the future. I think it can be written more clearly, e.g. how the behavior is examined and verified. Or skip the sentence completely perhaps?